### DISCUSSION TITLE
Unlock KeePassXC with fingerprint at login

### DISCUSSION THREAD

**Discussion by cristian-ene** [keepassxreboot/keepassxc#9290] [Q&A] [answered]
New user here, please don't throw stones if I missed it. 

I would like to instead of using the long DB password that I created, to use the fingerprint scanner on my laptop, or android device. 
If this is already a feature, can one please point me in the right direction?

**Comment by droidmonkey** [comment, marked as answer]
Permanent quick unlock is coming

**Reply by Plinsboorg** [reply]
Hi. Thank you for your work. Any idea how soon this feature will be available? I can see all relevant open issues are closed now, but I still can't see this feature in the actual app  #7020 #7935
Thank you

**Reply by droidmonkey** [reply]
https://github.com/keepassxreboot/keepassxc/tree/feature/remember-quickunlock

**Reply by Plinsboorg** [reply]
I assume this is the branch where this feature is implemented. I tried to build it on Windows using MINGW x64 to test the feature but after executing the "make" command I got the following error in the build process.
 [30%] Building CXX object src/CMakeFiles/keepassx_core.dir/gui/ApplicationSettingsWidget.cpp.obj
ApplicationSettingsWidget.cpp: In member function 'void ApplicationSettingsWidget::loadSetting
()':
ApplicationSettingsWidget.cpp:316:46: error: 'getQuickUnlock' was not declared in this scope
  316 |     m_secUi->quickUnlockCheckBox->setEnabled(getQuickUnlock()->isAvailable());

I got the same error on 2 different PCs when trying to build this branch. Is it supposed to be this way and the branch is still under development, or is this a problem on my end and it builds normally for other people? 

Is there some expected timeline for when this branch is planned to be merged with the main?

**Comment by vj-venkatesan** [comment]
Is the permanent unlock mechanism also coming for linux ?

**Reply by droidmonkey** [reply]
No, Linux doesn't have a **standard** secrets store that persists through reboots besides FDOSecrets, which we can replace. So it's complicated and likely not coming soon.

**Reply by stratozero** [reply]
I saw a fork `keepassxc-fprint`. Any plan to merge it (or to cherry-pick the key parts)?

**Reply by droidmonkey** [reply]
We already merged support

**Reply by stratozero** [reply]
Ok, probably my problem is that I'm using the flatpak version, and inside the keepassxc runtime `fprintd-*` is not present

**Reply by pascal06** [reply]
Hello,
sorry to go back this thread up, but Droidmonkey wrote that it has already merged support (about keepassxc-fprint): what does that mean?
Is there a working version of KeepassXC on linux that use fprint ?
I've got a "new" laptop (Dell Latitude 5580) that integrates a fprint reader: would I have a chance to use it with KeepassXC ?
Thanks a lot in advance,
Pascal

**Reply by droidmonkey** [reply]
You can read all about it here: https://github.com/keepassxreboot/keepassxc/pull/8983

Integration is manual at the moment, you need to copy the polkit policy file to the appropriate system directory to use this.

You can access a version that has this capability at https://snapshot.keepassxc.org

**Comment by bobdig** [comment]
Is it somehow possible to unlock the database with a Windows-Hello Fingerprintreader without allowing this Fingerprintreader to unlock the Windows Session? I want to be asked for a password any time a user tries to login into Windows.

**Reply by droidmonkey** [reply]
Not that I am aware of. Windows Hello is explicitly tied to your account unless you are on Active Directory and using Windows Hello for Business which is a different story.
